src/app/Layouts/defaultPageWrapper.jsx

```diff
@@ -1,51 +1,101 @@
-import React, { Fragment } from 'react';
+import React, { Fragment, useContext } from 'react';
 import { node, string, bool } from 'prop-types';
 import HeaderContainer from '../containers/Header';
 import FooterContainer from '../containers/Footer';
 import ManifestContainer from '../containers/Manifest';
 import ServiceWorkerContainer from '../containers/ServiceWorker';
-import { ServiceContextProvider } from '../contexts/ServiceContext';
+import {
+  ServiceContextProvider,
+  ServiceContext,
+} from '../contexts/ServiceContext';
 import { RequestContextProvider } from '../contexts/RequestContext';
 import ConsentBanner from '../containers/ConsentBanner';
 import getStatsDestination from '../contexts/RequestContext/getStatsDestination';
 import getStatsPageIdentifier from '../contexts/RequestContext/getStatsPageIdentifier';
 import getOriginContext from '../contexts/RequestContext/getOriginContext';
 import getEnv from '../contexts/RequestContext/getEnv';
 import GlobalStyle from '../lib/globalStyles';
+import deepGet from '../lib/utilities/deepGet';
+import getLangByPageType from '../contexts/RequestContext/getLangByPageType';
+
+const PageWithRequestContext = ({
+  children,
+  data,
+  env,
+  id,
+  isAmp,
+  isUK,
+  origin,
+  pageType,
+  service,
+}) => {
+  const { lang } = useContext(ServiceContext);
+  return (
+    <RequestContextProvider
+      env={env}
+      id={id}
+      isUK={isUK}
+      lang={getLangByPageType(data, lang, pageType)}
+      origin={origin}
+      pageType={pageType}
+      platform={isAmp ? 'amp' : 'canonical'}
+      statsDestination={getStatsDestination({
+        isUK,
+        env,
+        service,
+      })}
+      statsPageIdentifier={getStatsPageIdentifier({
+        pageType,
+        service,
+        id,
+      })}
+    >
+      <ServiceWorkerContainer />
+      <ManifestContainer />
+      <ConsentBanner />
+      <HeaderContainer />
+      {children}
+      <FooterContainer />
+    </RequestContextProvider>
+  );
+};
+
+PageWithRequestContext.propTypes = {
+  children: node.isRequired,
+  env: string.isRequired,
+  data: node.isRequired,
+  id: string,
+  isAmp: bool.isRequired,
+  isUK: bool.isRequired,
+  origin: string.isRequired,
+  pageType: string.isRequired,
+  service: string.isRequired,
+};
+
+PageWithRequestContext.defaultProps = {
+  id: null,
+};
 
 const PageWrapper = ({ bbcOrigin, children, id, service, isAmp, pageType }) => {
   const { isUK, origin } = getOriginContext(bbcOrigin);
   const env = getEnv(origin);
-
+  const data = deepGet(['props', 'data'], children);
```

Comment:
We shouldn't deepGet on the props of our children, There is a proper API to interact with children, but it would be preferable to ensure that PageWrapper is passed data directly

Reply:
@dr3 How would you pass data directly?

Reply:
Its already a prop to `defaultPageWrapper.jsx`, `props.data.pageData`

Reply:
Following https://github.com/bbc/simorgh/pull/2185 you will need to pass `pageData` to the request context and do your logic inside it